### DISCUSSION TITLE
Which compiler to use?

### DISCUSSION THREAD

**Discussion by notshriram** [Rinzii/genesis#59] [Q&A] [answered]
I am unable to compile the project with clang 16 because std::format is not supported yet, Is there any workaround for this or should I switch to gcc?

**Comment by Rinzii** [comment, marked as answer]
> I am unable to compile the project with clang 16 because std::format is not supported yet, Is there any workaround for this or should I switch to gcc?

Hello! Sorry for the long turn around on response. We actually compile the program in our CI using Clang 16 on Ubuntu 22.04. From the research we’ve done std::format should have complete support for all the features we are using with Clang 16. Could you provide information on your OS and what version of Clang you are compiling with? Also could you provide which branch you are compiling the project against?

Still we currently also support GCC 13 and MSVC using VS 22 so both of these are valid means of compiling the project if clang is being problematic!

Also thanks for trying out Genesis! I’ve been a bit busy with school but seeing others wanting to use the project always invigorates me to add more!
